### PR TITLE
fix staticcheck issues

### PR DESCRIPTION
**Pull request by gerardsn**
This is the first step towards adding [staticcheck](https://staticcheck.io) linter to the CI pipeline

### CODE REVIEW

File: network/transport/v2/senders.go [hunk 1]
```diff
@@ -91,7 +91,6 @@ func (p *protocol) sendTransactionListQuery(id transport.PeerID, refs []hash.SHA
 			Debug("Did not request a TransactionList while another conversation is in progress")
 		return nil
 	}
-	conversation.set("refs", refs)
```
Comment (woutslakhorst) [inline network/transport/v2/senders.go:94]: We do not check these anymore?

File: network/transport/grpc/outbound_connector.go [hunk 2]
```diff
@@ -150,7 +150,7 @@ func (c *outboundConnector) tryConnect() (*grpcLib.ClientConn, error) {
 	if c.tlsConfig != nil {
 		dialOptions = append(dialOptions, grpcLib.WithTransportCredentials(credentials.NewTLS(c.tlsConfig))) // TLS authentication
 	} else {
-		dialOptions = append(dialOptions, grpcLib.WithInsecure()) // No TLS, requires 'insecure' flag
+		dialOptions = append(dialOptions, grpcLib.WithTransportCredentials(insecure.NewCredentials())) // No TLS, requires 'insecure' flag
```
Comment (woutslakhorst) [inline network/transport/grpc/outbound_connector.go:153]: What is the reason for this change? Deprecation?

**Review by woutslakhorst — COMMENTED** [review]
Are there any instances of the 'append' fix that could benefit from a `copy`? Making sure the original slice is not returned?

File: network/transport/grpc/outbound_connector.go (hunk 2 above)
Comment (gerardsn) [reply network/transport/grpc/outbound_connector.go:153]: correct, this is the recommended replacement

File: network/transport/v2/senders.go (hunk 1 above)
Comment (gerardsn) [reply network/transport/v2/senders.go:94]: We do, but directly on the refs in the TransactionListQuery message that created conversation. The additional properties were unused/unchecked.

File: auth/services/x509/x509_validator_test_utils_test.go [hunk 3]
```diff
@@ -124,28 +123,3 @@ func createLeafCert(parent *x509.Certificate, caKey *rsa.PrivateKey) (*x509.Cert
 	cert, err := createTestCert(parent, template, &priv.PublicKey, caKey)
 	return cert, priv, err
 }
-
-func createCrl(currentCrl *x509.RevocationList, issuer *x509.Certificate, priv crypto.Signer, certsToRevoke []*x509.Certificate) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (gerardsn) [reply auth/services/x509/x509_validator_test_utils_test.go:128]: Good question. It looks like Crls are now handled in the `crl` package and this is package is used in the `x509_validator_test.go`

[the remark this answers is not in the document]
Comment (gerardsn) [reply auth/services/x509/x509_validator_test_utils_test.go:128]: I'm not familiar enough with the topic to determine the quality of testing

**Comment by gerardsn** [comment]
> Are there any instances of the 'append' fix that could benefit from a `copy`? Making sure the original slice is not returned?

made some adjustments to make sure new slices are returned where needed.

`b := make([]newSlice, len(a)); copy(b, a)` and `c := append([]newSlice, a...)` are both new slices that only differ in the capacity of the slice. https://go.dev/play/p/C2nPQFflsM2

**Review by woutslakhorst — APPROVED** [review]
